pennylane/io/qualtran_io.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the adapter class for Qualtran-PennyLane interoperability.
+"""
+# pylint:disable=
+
+import pennylane as qml
+from qualtran import (
+    Bloq,
+    CompositeBloq,
+    Soquet,
+    LeftDangle,
+    Side,
+    DecomposeNotImplementedError,
+    DecomposeTypeError,
+)
+from pennylane.operation import Operation
+from pennylane.wires import WiresLike
+
+import numpy as np
+
+def get_bloq_registers_info(bloq):
+    """Returns a `qml.registers` object associated with all named and unnamed registers and wires 
+    in the bloq.
+    
+    Args:
+        bloq: the bloq to get the registers info of
+
+    Returns:
+        dict: A dictionary that has all the named and un-named registers with default wire
+        ordering.
+
+    **Example**
+
+    Given a qualtran bloq:
+
+    from qualtran.bloqs.basic_gates import Swap
+
+    >>> qml.get_bloq_registers_info(Swap(3))
+    {'x': Wires([0, 1, 2]), 'y': Wires([3, 4, 5])}
```

Comment:
A bit confusing but what does `bitsize=3` in this case? Do we get a SWAP on three wires?

Reply:
The Bloq Swap basically swaps N bit registers. So in this case you would get `qml.SWAP([0, 3]), qml.SWAP([1. 4]), qml.SWAP([2, 5])`

Maybe SWAP is not the best example since the qualtran Swap is different from the QML Swap.